ql/src/java/org/apache/hadoop/hive/ql/parse/LoadSemanticAnalyzer.java

```diff
@@ -344,7 +358,7 @@ private void analyzeLoad(ASTNode ast) throws SemanticException {
     }
 
     // make sure the arguments make sense
-    List<FileStatus> files = applyConstraintsAndGetFiles(fromURI, ts.tableHandle);
+    List<FileStatus> files = applyConstraintsAndGetFiles(fromURI, ts.tableHandle, srcs);
```

Comment:
What is behaviour of below scenarios before and after fix and what exception it is throwing

Partitioned - Local path
Partitioned - HDFS path
Non-Partitioned - Local path
Non-Partitioned - HDFS path

Reply:
The changes were introduced only for Partition tables, there is no change in behaviour for Non Partition.
For partition tables for both local and hdfs path , Before the fix the exception thrown was File Not Found from Node Manager. After the fix the exception will be thrown from Hive Server itself, and the exception now would be  Semantic Exception.

Reply:
So earlier in partition case, even though file not exists query was successful. Can you check why it is not throwing earlier. 

After fix it was validated at server level and it will throw error for non existing file paths. So if any such cases are there in existing applications, those queries need to corrected.

Reply:
In the partition case if the file doesn't exist then number of splits will be zero, as the splits are zero the mappers also will be equal to zero. As there are no mappers the CombineHiveInputformat will not be called.

Now after the fix we would be validating at the server level, thus in all such cases in the existing applications, those queries needs to be corrected.

Reply:
> For partition tables for local ,
In Local path case, If file would be present at hs2 what is the behaviour after fix

Reply:
For partition case if the the file does exist at hs2, and the container looks for the file at other machines  it can throw the execution error from hdfs end, that scenario is not covered in this PR